File: src/leetcode/count-the-number-of-consistent-strings/count-the-number-of-consistent-strings.js
/**
 * Leetcode #1684
 * Difficulty: Easy
 * https://leetcode.com/problems/count-the-number-of-consistent-strings/
 */
/**
 * @param {string} allowed
 * @param {string[]} words
 * @return {number}
 */
export const countConsistentStrings = (allowed, words) => {
  const allowedSet = new Set();

  for (let i = 0; i < allowed.length; i += 1) {
    allowedSet.add(allowed[i]);
  }

  let count = 0;

  for (let i = 0; i < words.length; i += 1) {
    let isConsistent = true;

    for (let j = 0; j < words[i].length; j += 1) {
      if (!allowedSet.has(words[i][j])) {
        isConsistent = false;
        break;
      }
    }

    if (isConsistent) {
      count += 1;
    }
  }

  return count;
};

File: src/leetcode/count-the-number-of-consistent-strings/count-the-number-of-consistent-strings.rs
use std::collections::HashSet;

impl Solution {
    pub fn count_consistent_strings(allowed: String, words: Vec<String>) -> i32 {
        let mut count = 0;
        let set = allowed.chars().collect::<HashSet<_>>();
        
        for word in words.iter() {
            let mut isConsistent = true;
            
            for char in word.chars() {
                if !set.contains(&char) {
                    isConsistent = false;
                    break
                }
            }
            
            
            if (isConsistent) {
                count += 1;
            }
        }
        
        count
    }
}

// impl Solution {
//     pub fn count_consistent_strings(allowed: String, words: Vec<String>) -> i32 {
//         let mut consistent = 0;
//         for word in words.iter() {
//             if word.chars().all(|x| allowed.contains(x)) {
//                 consistent += 1;
//             }
//         }
//         consistent
//     }
// }